app/src/main/kotlin/be/zvz/kookie/Server.kt

```diff
@@ -272,6 +292,98 @@ class Server(dataPath: Path, pluginPath: Path) {
         TODO("Not yet implemented")
     }
 
+    private fun tickProcessor() {
+        nextTick = System.currentTimeMillis().toDouble()
+
+        while (isRunning) {
+            tick()
+        }
+    }
+
+    private fun tick() {
+        val tickTime: Long = System.currentTimeMillis()
+        if (tickTime - nextTick < -0.0025) {
+            return
+        }
+
+        // TODO: Timings.serverTick.startTiming()
+
+        ++tickCounter
+
+        Timings.schedulerSync.startTiming()
+        // TODO: pluginManager.tickSchedulers()
+        Timings.schedulerSync.startTiming()
+
+        // TODO: Timings.schedulerAsync.startTiming()
+        // TODO: asyncPool.collectTasks()
+        // TODO: Timings.schedulerAsync.stopTiming()
+
+        // TODO: worldManager.tick(tickCounter)
+
+        // TODO: Timings.connection.startTiming()
+        // TODO: network.tick()
+        // TODO: Timings.connection.stopTiming()
+        if ((tickCounter % 20) == 0L) {
+            if (doTitleTick) {
+                titleTick()
+            }
+            currentTPS = 20F
+            currentUse = 0F
+
+            val queryRegenerateEvent = QueryRegenerateEvent(QueryInfo(this))
+            queryRegenerateEvent.call()
+            queryInfo = queryRegenerateEvent.queryInfo
+
+            // TODO: network.updateName()
+            // TODO: network.bandwidthTracker.rotateAverageHistory()
+        }
+
+        if (sendUsageTicker > 0 && --sendUsageTicker == 0) {
+            sendUsageTicker = 6000
+            // TODO: sendUsage(SendUsageTask.Type.STATUS)
+        }
+
+        if ((tickCounter % 100) == 0L) {
```

Comment:
Two if statements which could be collapsed were detected. These statements can be merged to improve readability.